logic/transaction_pool.js

```diff
@@ -596,30 +599,29 @@ TransactionPool.prototype.processUnconfirmedTransaction = function(
 TransactionPool.prototype.queueTransaction = function(transaction, cb) {
 	transaction.receivedAt = new Date();
 
-	if (transaction.bundled) {
-		if (self.countBundled() >= config.transactions.maxTransactionsPerQueue) {
-			return setImmediate(cb, 'Transaction pool is full');
-		}
-		self.addBundledTransaction(transaction);
-	} else if (
-		transaction.type === transactionTypes.MULTI ||
-		Array.isArray(transaction.signatures)
-	) {
-		if (
-			self.countMultisignature() >= config.transactions.maxTransactionsPerQueue
+	__private.isMultisignatureTransaction(transaction, isMulti => {
+		if (transaction.bundled) {
```

Comment:
This way `isMultisignatureTransaction` will be called for every transaction, which is doing a DB read. While its necessary only if transaction is not bundled. So move this call after `transaction.bundled` condition to avoid DB calls without requirement.